cmd/uplink/cmd/cp.go

```diff
@@ -159,21 +139,21 @@ func download(ctx context.Context, bs buckets.Store, srcObj *url.URL, destFile s
 		bar.Finish()
 	}
 
-	if destFile != "-" {
-		fmt.Printf("Downloaded %s to %s\n", srcObj, destFile)
+	if destObj.Base() != "-" {
+		fmt.Printf("Downloaded %s to %s\n", srcObj.String(), destObj.String())
 	}
 
 	return nil
 }
 
 // copy copies s3 compatible object args[0] to s3 compatible object args[1]
-func copy(ctx context.Context, bs buckets.Store, srcObj *url.URL, destObj *url.URL) error {
-	o, err := bs.GetObjectStore(ctx, srcObj.Host)
+func copy(ctx context.Context, bs buckets.Store, srcObj fpath.FPath, destObj fpath.FPath) error {
```

Comment:
Let's add a guard here for checking for checking:
```
if srcFile.IsLocal() || destObj.IsLocal() { return some error }
```

Reply:
Done.